cpg_utils/hail_batch.py

```diff
@@ -606,3 +607,45 @@ def query_command(
 EOT
 python3 script.py
 """
+
+
+def start_query_context(
+    query_backend: typing.Literal['spark', 'batch', 'local', 'spark_local']
+    | None = None,
+    log_path: str | None = None,
+    dataset: str | None = None,
+    billing_project: str | None = None,
+):
+    """
+    Start Hail Query context, depending on the backend class specified in
+    the hail/query_backend TOML config value.
+    """
+    query_backend = query_backend or get_config().get('hail', {}).get(
+        'query_backend', 'spark'
+    )
+    if query_backend == 'spark':
+        hl.init(default_reference=genome_build())
+    elif query_backend == 'spark_local':
+        hl.init(
+            default_reference=genome_build(),
+            master='local[2]',
```

Comment:
Is `local[2]` deliberate or should this be `*`?

```suggestion
            master='local[*]',
```

Reply:
`2` means "run spark locally with 2 threads". `*` would mean "run with all threads available" https://stackoverflow.com/questions/32356143/what-does-setmaster-local-mean-in-spark which is not a good practice for local tests, I think

Reply:
I'll clarify that in the code

Reply:
@lgruen added a constant for the number of threads, and a comment